src/components/editor/ImportResourceTemplate.jsx

```diff
@@ -1,120 +1,123 @@
-        this.updateStateForResourceError(error)
+  createResource = async (content, group) => {
+    try {
+      const response = await this.instance.createResourceWithHttpInfo(group, content, { slug: content.id, contentType: 'application/json' })
+      return response.response
+    } catch(error) {
+      this.setState({
+        createResourceError: [...this.state.createResourceError, error.response]
       })
-    )
-  }).then(response => {
-    if(this.state.createResourceError.length === profileCount) {
-      this.setState({modalShow: true})
+      return error.response
     }
-    return response
-  }).catch(() => {})
-
-  updateResourcePromise = (content, group) => new Promise(async (resolve) => {
-    resolve(
-      await instance.updateResourceWithHttpInfo(group, content.id, content, { contentType: 'application/json' }).catch((error) => {
-        this.setState({
-          updateResourceError: error
-        })
-      })
-    )
-  }).then(response => {
-
-    return response
-  }).catch(() => {})
-
-  fulfillCreateResourcePromise = async (promise) => {
-    await promise.then(result => {
-      const joinedMessages = this.state.message.slice(0)
-
-      if(result.response.statusText !== 'No Content') {
-        this.setState({tempState: `${result.response.statusText} ${result.response.headers.location}`})
-        joinedMessages.push(this.state.tempState)
-        this.setState({message: joinedMessages})
-      }
-
-    }).catch(() => {
-      if (this.state.createResourceError[0].statusText !== 'Conflict') {
-        const msg = 'The sinopia server is not accepting the request for this resource.'
-        this.state.updateResourceError ? this.setState({message: [`${msg}: ${this.state.updateResourceError}`]}) : this.setState({message: [msg]})
-      }
-    })
   }
 
-  updateStateForResourceError = (error) => {
-    if(_.get(error, 'response.statusText') === 'Conflict') {
-      const joinedConflicts = [...this.state.createResourceError]
-      joinedConflicts.push(error.response)
-      this.setState({createResourceError: joinedConflicts})
+  updateResource = async (content, group) => {
+    try {
+      const response = await this.instance.updateResourceWithHttpInfo(group, content.id, content, { contentType: 'application/json' })
+      return response.response
+    } catch(error) {
+      this.setState({ updateResourceError: error.response })
+      return error.response
     }
   }
 
-  updateAllResources = (rts, group) => {
-    return Promise.all(rts.map(async rt =>
-      await this.fulfillCreateResourcePromise(this.updateResourcePromise(rt, group))
-    ))
+  updateStateFromServerResponse = (response, operation, profileCount) => {
+    // HTTP status 409 == Conflict
+    if (operation == 'update' && response.status !== 409) {
+      const msg = 'The sinopia server is not accepting the request for this resource.'
+      this.state.updateResourceError ?
```

Comment:
👍  on this multiple-line conditional, I think it reads easier than having this on a single line.